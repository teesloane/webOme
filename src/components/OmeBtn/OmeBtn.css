.OmeBtn {
  display: flex;
  width: 50px;
  height: 50px;
  background: aliceblue;
  border: 2px solid black;
  border-radius: 3px;
  cursor: pointer;
  margin: 0.25rem;
  justify-content: center;
  align-items: center;
}

.OmeBtn-on {
  background: #3675CD;
}

.OmeBtn-on-glow {
  background: #70A6F2;
  box-shadow: 0px 0px 5px #3675CD;

}

.OmeBtn-Note-Hover {
  opacity: 0;
}

.OmeBtn:hover .OmeBtn-Note-Hover{
  opacity: 0.5;
}