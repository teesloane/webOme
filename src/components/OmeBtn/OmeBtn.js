import React from 'react';
import {observer} from 'mobx-react'
import './OmeBtn.css';
import OmeStore from '../../stores/OmeStore'
import { midiToNote } from '../../music_constants/index'

console.log(midiToNote)
/**
 * @description OmeStore button. Can turn on or off.
 * Interaction: onClick -> set it's state reference to it's inverse
 * TODO: Probably should refactor the onClick this to a method on the OmeStore class.
 */
var OmeBtn = observer(function OmeBtn(props) {
  const {rowId, noteId} = props
  let isPlaying = OmeStore.midiNotes[rowId][noteId].isPlaying

  return (
    <main 
      className={`OmeBtn ${isPlaying ? 'OmeBtn-on' : ''}`}
      onClick={() => OmeStore.midiNotes[rowId][noteId].isPlaying = !OmeStore.midiNotes[rowId][noteId].isPlaying } 
    >
      <span className="OmeBtn-Note-Hover">
        {midiToNote[props.note.midiNote].slice(0, -1)}
      </span>
    </main> 
  )
})

export default OmeBtn
