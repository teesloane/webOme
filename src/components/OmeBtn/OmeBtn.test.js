import OmeBtn from './OmeBtn'

// this is broken until stores are passed as props. 
// Can't easily test when stores are just "imported" :(
describe('omeBtn component', function() {
  beforeEach(function() {
    this.omeStore = {}
  })

  it('renders', () => {
    shallow(<OmeBtn omeStore={this.omeStore} />);
  })

  it('matches snapshot', () => {
    const component = shallow(<OmeBtn omeStore={this.omeStore} />);
    expect(component).toMatchSnapshot();
  })

})