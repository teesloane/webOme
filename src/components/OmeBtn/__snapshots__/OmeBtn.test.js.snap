exports[`omeBtn component matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <OmeBtn
    omeStore={undefined} />,
  "nodes": Array [
    <OmeBtn
      omeStore={undefined} />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <inject-OmeBtn-with-omeStore
        omeStore={undefined} />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance":  {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "mobxStores": undefined,
        },
        "props": Object {
          "omeStore": undefined,
        },
        "refs": Object {},
        "state": null,
        "storeRef": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <OmeBtn
          omeStore={undefined} />,
        "_debugID": 4,
        "_renderedOutput": <OmeBtn
          omeStore={undefined} />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <inject-OmeBtn-with-omeStore
    omeStore={undefined} />,
}
`;
